xds/googledirectpath/googlec2p.go

```diff
@@ -166,30 +160,45 @@ func (r *c2pResolver) Close() {
 	r.clientCloseFunc()
 }
 
-var ipv6EnabledMetadata = &structpb.Struct{
-	Fields: map[string]*structpb.Value{
-		ipv6CapableMetadataName: structpb.NewBoolValue(true),
-	},
-}
-
 var id = fmt.Sprintf("C2P-%d", grpcrand.Int())
 
-// newNode makes a copy of defaultNode, and populate it's Metadata and
-// Locality fields.
-func newNode(zone string, ipv6Capable bool) *v3corepb.Node {
-	ret := &v3corepb.Node{
-		// Not all required fields are set in defaultNote. Metadata will be set
-		// if ipv6 is enabled. Locality will be set to the value from metadata.
-		Id:                   id,
-		UserAgentName:        gRPCUserAgentName,
-		UserAgentVersionType: &v3corepb.Node_UserAgentVersion{UserAgentVersion: grpc.Version},
-		ClientFeatures:       []string{clientFeatureNoOverprovisioning},
-	}
-	ret.Locality = &v3corepb.Locality{Zone: zone}
+func newNodeConfig(zone string, ipv6Capable bool) string {
+	metadata := ""
 	if ipv6Capable {
-		ret.Metadata = ipv6EnabledMetadata
+		metadata = fmt.Sprintf(`, "metadata":  { "%s": true }`, ipv6CapableMetadataName)
 	}
-	return ret
+	return fmt.Sprintf(`
+	{
+		"id": "%s",
+		"user_agent_name": "%s",
+		"UserAgentVersionType": {
+			"userAgentVersion": "%s"
+		},
+		"client_features": ["%s", "%s"],
```

Comment:
`UserAgentVersionType` is in camel case while the rest seem to be in snake case. 


But this got me thinking `newConfigFromContents` overrides these values here https://github.com/grpc/grpc-go/blob/cce163274b6cb9de2b2bf0bc742384e5755fee42/xds/internal/xdsclient/bootstrap/bootstrap.go#L568-L570


I would prefer if we didnt set it here. Sorry about the back and forth on this.

```suggestion
```